src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -91,5 +95,33 @@ private int GetServerMinMemory()
         {
             return configService.GetServerSmoConfig(server, configService.MinServerMemoryPropertyNumber).ConfigValue;
         }
+
+        private void UpdateServerProperties(InitializeViewRequestParams viewParams, ServerInfo serverInfo)
+        {
+            if (viewParams != null)
+            {
+                ConnectionInfo connInfo = this.GetConnectionInfo(viewParams.ConnectionUri);
+                ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+                CDataContainer dataContainer = CDataContainer.CreateDataContainer(connInfo);
+
+                ServerPrototype prototype = new ServerPrototype(dataContainer);
+                prototype.ApplyInfoToPrototype(serverInfo);
+                ConfigureServerRole(dataContainer, ConfigAction.Update, RunType.RunNow, prototype);
+            }
+        }
+
+        private void ConfigureServerRole(CDataContainer dataContainer, ConfigAction configAction, RunType runType, ServerPrototype prototype)
+        {
+            string sqlScript = string.Empty;
+            using (var actions = new ServerActions(dataContainer, prototype, configAction))
+            {
+                var executionHandler = new ExecutonHandler(actions);
+                executionHandler.RunNow(runType, this);
+                if (executionHandler.ExecutionResult == ExecutionMode.Failure)
+                {
+                    throw executionHandler.ExecutionFailureException;
```

Comment:
Is this going to have information about why it failed?

Reply:
I checked and this is the same implementation that ServerRoleHandler and DatabaseHandler follow. Should I append a more descriptive error to the exception data property or create a new exception?

Reply:
Looks like this property is just a cached version of the last thrown exception, so it will still have any relevant error messages.